pkg/compare/strcompare.go

```diff
@@ -44,6 +44,9 @@ func (c *strCompare) Compare(veci, vecj int, vi, vj int64) int {
 	if cmp != 0 {
 		return cmp
 	}
+	if nulls.Contains(c.vs[veci].Nsp, uint64(vi)) && nulls.Contains(c.vs[veci].Nsp, uint64(vj)) {
```

Comment:
Are you saying null = null?    Why nullsCompare does not do the right thing?

Reply:
nullsCompare will treat both null as equal, it returns 0, and for T_any type, the data attribute in the vector is nil, not like other type, it will have default data. So we need to treat this, if don't add this , it will result nil pointer error.